feat(ISV-6497): use pullspecs with digests in package metadata

jedinym opened the pull request:
Resolved pullspecs from containerfile into pullspecs with digests. We need this to properly match package origins with image SPDX packages, as those only contain digests.

Also added a new CopyType field for parsed COPY commands, to disambiguate the `from` field (which can be either a stage alias or a pullspec for builder and external copies respectively).

jedinym committed:
feat: specify copy type in parsed containerfile

The CopyType enum helps distinguish between multiple possible values of
the Copy.From field, which can be either a builder stage or a pullspec.

Signed-off-by: Martin Jediny <jedinym@proton.me>

jedinym committed:
feat: use pullspecs with digests in package metadata

Signed-off-by: Martin Jediny <jedinym@proton.me>

BorekZnovustvoritel reviewed (approved):
LGTM

ezopezo commented on pkg/containerfile/containerfile.go:
```diff
@@ -134,6 +146,8 @@ func mapAliasesToPullspecs(stages []imagebuilder.Stage) map[string]string {
 // present in that stage.
 // A COPY command is builder-type if the "--from" flag is specified and it copies from
 // a builder stage or directly from an image.
+// Uses the passed previouse stageNames to specify whether copies are from a stage
```
```suggestion
// Uses the passed previous stageNames to specify whether copies are from a stage
```

ezopezo commented on pkg/scan.go:
```diff
@@ -103,9 +117,72 @@ func Scan(
+func resolvePullspecs(store storage.Store, stages []containerfile.Stage) (map[string]string, error) {
+	res := make(map[string]string)
+
+	for _, stage := range stages[:len(stages)-1] {
```
Doesn't this prevent resolving pullspecs for the external images referenced in final stage?

jedinym replied:
Great catch, will fix.

jedinym committed:
fix: resolve copy pullspecs also in final stage

Signed-off-by: Martin Jediny <jedinym@proton.me>

jedinym committed:
docs: fix typo

Signed-off-by: Martin Jediny <jedinym@proton.me>

jedinym commented:
/retest

jedinym committed:
test: use resolved pullspec in integration tests

Signed-off-by: Martin Jediny <jedinym@proton.me>

jedinym committed:
lint: minor suggestion fix

Signed-off-by: Martin Jediny <jedinym@proton.me>

jedinym committed:
test: don't match pullspecs exactly in integration tests

Digests of images are unreliable and can change between different
environments. We can only compare the part of a pullspec without the
digest.

Signed-off-by: Martin Jediny <jedinym@proton.me>

jedinym committed:
chore: add snyk ignore file

Signed-off-by: Martin Jediny <jedinym@proton.me>
